dt-posts/dt-posts.php

```diff
@@ -1603,154 +1597,34 @@ private static function advanced_search_by_post( string $post_type, array $query
-                                    AND REPLACE( meta_value, ' ', '') LIKE '%" . esc_sql( str_replace( ' ', '', $search ) ) . "%'
-                        )
-                    ";
-                } else {
-                    $post_query .= "AND ( ";
-                }
-                if ( in_array( 'all', $fields_to_search ) ) {
-                    if ( substr( $post_query, - 6 ) !== 'AND ( ' ) {
-                        $post_query .= "OR ";
-                    }
-                    $post_query .= "( p.post_title LIKE '%" . esc_sql( $search ) . "%' ) OR p.ID IN ( SELECT comment_post_ID
-                    FROM $wpdb->comments
-                    WHERE comment_content LIKE '%" . esc_sql( $search ) . "%'
-                    ) OR p.ID IN ( SELECT post_id
-                    FROM $wpdb->postmeta
-                    WHERE meta_value LIKE '%" . esc_sql( $search ) . "%'
-                    ) ";
-                } else {
-                    if ( in_array( 'comment', $fields_to_search ) ) {
-                        if ( substr( $post_query, - 6 ) !== 'AND ( ' ) {
-                            $post_query .= "OR ";
-                        }
-                        $post_query .= " p.ID IN ( SELECT comment_post_ID
-                        FROM $wpdb->comments
-                        WHERE comment_content LIKE '%" . esc_sql( str_replace( ' ', '', $search ) ) . "%'
-                        ) ";
-                    }
-                    foreach ( $fields_to_search as $field ) {
-                        array_push( $other_search_fields, $field );
-                    }
-                }
-            }
-            foreach ( $other_search_fields as $field ) {
-                if ( substr( $post_query, - 6 ) !== 'AND ( ' ) {
-                    $post_query .= "OR ";
-                }
-                $post_query .= "p.ID IN ( SELECT post_id
-                             FROM $wpdb->postmeta
-                             WHERE meta_key LIKE '" . esc_sql( $field ) . "'
-                             AND meta_value LIKE '%" . esc_sql( $search ) . "%'
-                ) ";
-            }
-            $post_query .= " ) ";
-
-            if ( $post_type === "peoplegroups" ) {
-
-                $locale = get_user_locale();
-
-                $post_query = " AND (p.post_title LIKE '%" . esc_sql( $search ) . "%' OR p.ID IN ( SELECT post_id
-                                  FROM $wpdb->postmeta
-                                  WHERE meta_key LIKE '" . esc_sql( $locale ) . "'
-                                  AND meta_value LIKE '%" . esc_sql( $search ) . "%' ))";
-            }
-        }
-
-        $sort_sql = "";
-        if ( empty( $sort_sql ) ) {
-            $sort_sql = "p.post_title asc";
-        }
-
-        $group_by_sql = "";
-        if ( strpos( $sort_sql, 'sort.meta_value' ) !== false ) {
-            $group_by_sql = ", sort.meta_value";
-        }
-
-        $permissions = [
```

Comment:
Hey @kodinkat 
I think we still need this permission checking section
Otherwise the search with return results of posts the user does not have access to.

Reply:
ah, ok; let me just get eyes on...